src/internal/components/dropdown/__tests__/dropdown-position.test.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { applyDropdownPositionRelativeToViewport } from '../../../../../lib/components/internal/components/dropdown/dropdown-position';
+
+describe('applyDropdownPositionRelativeToViewport', () => {
+  const triggerRect = {
+    blockSize: 50,
+    inlineSize: 100,
+    insetBlockStart: 100,
+    insetInlineStart: 100,
+    insetBlockEnd: 150,
+    insetInlineEnd: 200,
+  };
+
+  const baseDropdownPosition = {
+    blockSize: '100px',
+    inlineSize: '100px',
+    insetInlineStart: '100px',
+    dropBlockStart: false,
+    dropInlineStart: false,
+  };
+
+  test("sets block end when the dropdown is anchored to the trigger's block start (expands up)", () => {
+    const dropdownElement = document.createElement('div');
+    applyDropdownPositionRelativeToViewport({
+      dropdownElement,
+      triggerRect,
+      position: { ...baseDropdownPosition, dropBlockStart: true },
+      isMobile: false,
+    });
+    expect(dropdownElement.style.insetBlockEnd).toBeTruthy();
```

Comment:
In a couple cases (here and 3 tests further down) I'm testing just that a value is set (truthy), instead of the actual value. This is because the value is a CSS expression using `calc()`, and I don't think it makes sense to be so specific about the implementation. We already have visual regression tests for the expected user-facing end result, which is what matters.